pkg/gardenlet/controller/federatedseed/seed_control.go

```diff
@@ -215,48 +238,95 @@ func (f *federatedSeedControllerManager) createControllers(ctx context.Context,
 	}
 
 	var (
-		childCtx, cancelFunc = context.WithCancel(ctx)
-		seedLogger           = logger.Logger.WithField("Seed", seedName)
+		extensionCtx, extensionCancelFunc = context.WithCancel(ctx)
+		dnsInformers                      = dnsinformers.NewSharedInformerFactory(dnsClient, 0)
+		extensionsInformers               = extensionsinformers.NewSharedInformerFactory(extensionsClient, 0)
+		extensionsController              = extensions.NewController(ctx, k8sGardenClient, k8sSeedClient, seedName, dnsInformers, extensionsInformers, seedLogger, recorder)
+	)
+
+	seedLogger.Info("Run extensions controller")
+	if err := extensionsController.Run(extensionCtx, *config.Controllers.ControllerInstallationRequired.ConcurrentSyncs, *config.Controllers.ShootStateSync.ConcurrentSyncs); err != nil {
+		seedLogger.Infof("There was an error running the extension controllers: %v", err)
+		extensionCancelFunc()
+		return err
+	}
+
+	f.addExtensionController(seedName, extensionsController, extensionCancelFunc)
+	return nil
+}
+
+func (f *federatedSeedControllerManager) createNamespaceControllers(ctx context.Context, seedLogger *logrus.Entry, seedName string, k8sGardenClient kubernetes.Interface, config *config.GardenletConfiguration, recorder record.EventRecorder) error {
+	if _, found := f.namespaceControllers[seedName]; found {
+		logger.Logger.Infof("Namespace controllers are already started for seed %s", seedName)
+		return nil
+	}
+
+	k8sSeedClient, err := seedpkg.GetSeedClient(ctx, k8sGardenClient.Client(), config.SeedClientConnection.ClientConnectionConfiguration, config.SeedSelector == nil, seedName)
+	if err != nil {
+		return err
+	}
 
-		dnsInformers         = dnsinformers.NewSharedInformerFactory(dnsClient, 0)
-		extensionsInformers  = extensionsinformers.NewSharedInformerFactory(extensionsClient, 0)
-		extensionsController = extensions.NewController(ctx, k8sGardenClient, k8sSeedClient, seedName, dnsInformers, extensionsInformers, seedLogger, recorder)
+	var (
+		namespaceCtx, namespaceCancelFunc = context.WithCancel(ctx)
+		seedKubeInformerFactory           = kubeinformers.NewSharedInformerFactory(k8sSeedClient.Kubernetes(), 0)
+		seedDefaultNamespaceKubeInformer  = kubeinformers.NewSharedInformerFactoryWithOptions(k8sSeedClient.Kubernetes(), 0, kubeinformers.WithNamespace(corev1.NamespaceDefault))
```

Comment:
It's not a good practice to have more than 1 informer factory (more memory consumption). Write a comment that if some other controller would need to have access to the `Endpoints` informer in the Seed cluster that this network policy controller should be changed to filter for the `default` namespace and reuse the kubenetes informer factory without options.

Reply:
To avoid accidental breakage in the future I would suggest to filter for `default` namespace now for `Endpoints` event filter.

Reply:
I have added an explanatory comment. However I did not add an additional filter function, but explained what needs to be done if the current informer factory would be replaced with an informer factory that is not already filtering for the default namespace.